### ISSUE TITLE
Update `GlobalTransform` as part of `FixedUpdate`

### ISSUE THREAD

**Issue by jabuwu** [bevyengine/bevy#7836]
## What problem does this solve or what need does it fill?

The purpose of `FixedUpdate` and fixed timesteps in general is to improve determinism in systems, but any fixed update system relying on `GlobalTransform` will currently suffer from a lack of determinism if it runs multiple times in 1 frame.

## What solution would you like?

`GlobalTransform` should be updated between fixed updates.

## What alternative(s) have you considered?

Games could add these systems to `FixedUpdate` themselves. It's actually fairly easy to do already.

```
use bevy::{
    prelude::*,
    transform::systems::{propagate_transforms, sync_simple_transforms},
};

#[derive(Debug, Hash, PartialEq, Eq, Clone, SystemSet)]
pub enum FixedTransformSystem {
    TransformPropagate,
}

// ...

app.add_system_to_schedule(
    CoreSchedule::FixedUpdate,
    sync_simple_transforms.in_set(FixedTransformSystem::TransformPropagate),
)
.add_system_to_schedule(
    CoreSchedule::FixedUpdate,
    propagate_transforms.in_set(FixedTransformSystem::TransformPropagate),
);
```

**Comment by alice-i-cecile** [comment]
WRT transform propagation, @inodentry has previously advocated for separating out a "physics transform" from a "rendering transform", and using interpolation to avoid visual hiccups.

This is pretty compelling to me, but definitely doesn't seem compatible with the approach proposed here.

**Comment by alice-i-cecile** [comment]
I would also be interested in seeing if https://github.com/bevyengine/bevy/issues/7838 could be used to resolve these problems more gracefully.

**Comment by jabuwu** [comment]
> I would also be interested in seeing if #7838 could be used to resolve these problems more gracefully.

While writing this issue, I did think about suggesting a way to manually "flush" `GlobalTransform` the same way we do command buffers, but quickly realized this is already possible. It's just not very pretty. Your `before_and_seperated_by` syntax is a nice solution here. I think it needs to be `before_and_seperated_by(b, (sync_simple_transforms, propagate_transforms))`. With two systems?

**Comment by alice-i-cecile** [comment]
Yeah agreed: that's a nice catch that we may need to force separation by a collection of systems instead.

**Comment by james7132** [comment]
In it's current state, transform propagation requires a full hierarchy traversal, this can be quite expensive, even if nothing changes, on large worlds. While this is probably needed for ECS-native physics simulation to work, I'm hesitant to suggest adding it now when the FixedUpdate schedule is empty by default, and runs potentially multiple times per frame. Ideally we should have some optimizations like #7840 before we enable this.

**Comment by JoJoJet** [comment]
IMO this isn't controversial, just not feasible performance-wise for now.

**Comment by BenjaminBrienen** [comment]
@JoJoJet is this still blocked?

**Comment by JoJoJet** [comment]
As far as I know yes, the transform propagation algorithm hasn't been changed since posting this. In order to alleviate the performance concerns we'd need two changes:

1. Update `propogate_transforms` to only visit the transforms of entities that have changed. Currently we visit the entire hierarchy on every run of the system which is very inefficient (this is sub-optimal regardless of whether we update transforms in FixedUpdate, but this issue would exacerbate it)
2. We'd need a way of sharing change ticks between multiple instances of `propogate_transforms`. Currently, if you were to add `propogate_transforms` to the schedule multiple times, each instance of the system would have its own change ticks, which means each instance would redundantly propagate transforms for the same entities/components

**Comment by JoJoJet** [comment]
See also https://github.com/bevyengine/bevy/pull/9497, which updated the visibility propagation algorithm to only visit entries that were changed. That exact approach probably would not be ideal for transforms though, since it can result in redundant propagations for a single entity within the same frame. For visibility, we figured that the tradeoff was worth it since propagating visibility is relatively cheap, and the cost of visiting the entire hierarchy tends to be more expensive than propagating changes entities. Transform propagation is computationally more expensive so it's not clear if that same tradeoff would be worth it.
